### PR TITLE
Remove Parse Trimming Logic

### PR DESCRIPTION
**Pull request by Maithem**
## Overview
We currently only support prefix trimming, removed all the logic
related to sparse trimming, which will also prevent the related
metadata files from being created.

Why should this be merged: Prevents unnecessary metadata from being generated. 

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java [hunk 1]
```diff
@@ -223,14 +223,6 @@ private void fillHole(CorfuPayloadMsg<FillHoleRequest> msg, ChannelHandlerContex
...
-    @ServerHandler(type = CorfuMsgType.TRIM)
-    private void trim(CorfuPayloadMsg<TrimRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
-        TrimRequest req = msg.getPayload();
-        batchWriter.trim(req.getAddress().getSequence(), req.getAddress().getEpoch());
-        //TODO(Maithem): should we return an error if the write fails
-        r.sendResponse(ctx, msg, CorfuMsgType.ACK.msg());
-    }
-
     @ServerHandler(type = CorfuMsgType.PREFIX_TRIM)
     private void prefixTrim(CorfuPayloadMsg<TrimRequest> msg, ChannelHandlerContext ctx,
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java]: Shouldn't we remove the trim logic from the LogUnitClient as well?
And also clean up, by removing the trim CorfuMsgType.

File: test/src/test/java/org/corfudb/infrastructure/log/StreamLogFilesTest.java [hunk 2]
```diff
@@ -543,7 +453,7 @@ public void testPrefixTrim() {
 
         // Write 50 segments and trim the first 25
         final long numSegments = 50;
-        final long filesPerSegment = 3;
+        final long filesPerSegment = 1;
```
Comment (annym) [inline test/src/test/java/org/corfudb/infrastructure/log/StreamLogFilesTest.java]: why did you change this?

File: infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java (hunk 1 above)
Comment (Maithem) [reply infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java]: Done.

File: test/src/test/java/org/corfudb/infrastructure/log/StreamLogFilesTest.java (hunk 2 above)
Comment (Maithem) [reply test/src/test/java/org/corfudb/infrastructure/log/StreamLogFilesTest.java]: Currently on master, we have 3 files per segment. For example, for segment 1, we have 1.log          1.log.pending  and 1.log.trimmed. This test used to verify that the 3 files were being generated. Now that we removed those 2 files, the assertion would fail.
